sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/BasicDigitalTwinMetadata.cs

```diff
@@ -19,5 +20,12 @@ public class BasicDigitalTwinMetadata
         /// </summary>
         [JsonPropertyName(DigitalTwinsJsonPropertyNames.MetadataModel)]
         public string ModelId { get; set; }
+
+        /// <summary>
+        /// This field will contain metadata about changes on properties on the digital twin.
+        /// For your convenience, the value can be deserialized into <see cref="BasicDigitalTwinPropertyMetadata"/>.
+        /// </summary>
+        [JsonExtensionData]
```

Comment:
AFAIK - the limitation driving IDictionary<string, object> is a limitation in JsonExtensionData. Did we consider providing a custom attribute on BasicDigitalTwinMetadata which would know how to deserialize these elements? This would allow us to expose

IDictionary<string, DigitalTwinPropertyMetadata>.

And eliminate a bunch of serialization converting that customers would need to do.

Reply:
Something like:
```
using System;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.InteropServices;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace ConsoleApp6
{
    [JsonConverter(typeof(DigitalTwinMetadataJsonConverter))]
    class DigitalTwinMetadata
    {
        public string ModelId { get; set; }

        public IDictionary<string, DigitalTwinPropertyMetadata> PropertyMetadata = new Dictionary<string, DigitalTwinPropertyMetadata>();
    }

    class DigitalTwinPropertyMetadata
    {
        [JsonPropertyName("lastUpdateTime")]
        public DateTimeOffset? LastUpdateTime { get; set; }
    }

    class DigitalTwinMetadataJsonConverter : JsonConverter<DigitalTwinMetadata>
    {
        public override DigitalTwinMetadata Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
        {
            if (reader.TokenType == JsonTokenType.Null)
            {
                return null;
            }
            else if (reader.TokenType == JsonTokenType.StartObject)
            {
                var metadata = new DigitalTwinMetadata();
                reader.Read(); // Advance into our object.
                while (reader.TokenType != JsonTokenType.EndObject) // Until we reach the end of the object we began reading
                {
                    if (reader.TokenType == JsonTokenType.PropertyName)
                    {
                        if (reader.GetString() == "$model")
                        {
                            reader.Read(); // advance to the string
                            metadata.ModelId = JsonSerializer.Deserialize<string>(ref reader, options);
                            reader.Read(); // Consume the string we deserialized
                        }
                        else
                        {
                            var propertyName = reader.GetString();
                            reader.Read(); // Should place us on StartObject
                            if (reader.TokenType == JsonTokenType.StartObject)
                            {
                                metadata.PropertyMetadata[propertyName] = JsonSerializer.Deserialize<DigitalTwinPropertyMetadata>(ref reader, options);
                                reader.Read(); // Consume EndObject
                            }
                        }
                    }
                }
                return metadata;
            }
            else
            {
                throw new Exception($"Unknown token type {reader.TokenType} at index {reader.TokenStartIndex}.");
            }
        }

        public override void Write(Utf8JsonWriter writer, DigitalTwinMetadata value, JsonSerializerOptions options)
        {
            writer.WriteStartObject();
            writer.WriteString("$model", value.ModelId);
            foreach(var p in value.PropertyMetadata)
            {
                writer.WritePropertyName(p.Key);
                JsonSerializer.Serialize<DigitalTwinPropertyMetadata>(writer, p.Value, options);
            }
            writer.WriteEndObject();
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            var metadata = new DigitalTwinMetadata
            {
                ModelId = "mymodel",
                PropertyMetadata = new Dictionary<string, DigitalTwinPropertyMetadata>()
                {
                    { "p1", new DigitalTwinPropertyMetadata { LastUpdateTime = DateTimeOffset.Now } },
                    { "p2", new DigitalTwinPropertyMetadata { LastUpdateTime = DateTimeOffset.Now.Subtract(TimeSpan.FromDays(1)) } },
                }
            };

            var ser = JsonSerializer.Serialize(metadata);
            Console.WriteLine(ser);

            // round trip
            var roundTripped = JsonSerializer.Deserialize<DigitalTwinMetadata>(ser);
            Console.WriteLine($"ModelId - {roundTripped.ModelId}");
            foreach(var v in roundTripped.PropertyMetadata)
            {
                Console.WriteLine($"{v.Key} - LastUpdated @ {v.Value.LastUpdateTime}");
            }
        }
    }
}
```

Reply:
We can't control which json serialization settings the user uses.

Reply:
How is attributing with a JsonConverter any different than attributing with JsonExtensionData? In either case, the owner of the JsonSerializerOptions can configure the set of JsonConverter objects that are used for any type... this is just controlling the defaults, and providing the best possible default seems like it would have value.